flink-runtime/src/main/java/org/apache/flink/runtime/operators/shipping/OutputEmitter.java

```diff
@@ -231,4 +268,44 @@ private int murmurHash(int k) {
 			throw new RuntimeException("Error while calling custom partitioner.", t);
 		}
 	}
+
+	private final int compareRecordAndBoundary(T record, Object[] boundary) {
+		TypeComparator[] flatComparators = this.comparator.getFlatComparators();
```

Comment:
can we cache the `flatComparators` and `keys` to avoid many object instantiations?